formatter.go

```diff
@@ -14,33 +15,62 @@ import (
 // the write entry
 type Formatter interface {
 	Format(w io.Writer, logLevel LogLevel, msg string, timestamp time.Time, fields Fields) error
+	AddMapping(fieldMapping FieldMapping)
 }
 
 // JSONFormatter used to output logs in JSON format
 type JSONFormatter struct {
-	Compact bool
+	FieldMapping FieldMapping
+	Compact      bool
 }
 
 func (j JSONFormatter) getKey(key string) string {
 	if j.Compact {
-		return "@" + key[:1]
+		var mappedKey string
+		var ok bool
+
+		if mappedKey, ok = j.FieldMapping[key]; !ok {
+			mappedKey = key
+		}
+
+		return mappedKey
 	}
 	return key
 }
 
+func (j *JSONFormatter) AddMapping(fieldMapping FieldMapping) {
+	var mapping = make(FieldMapping, 3)
+
+	// first add the custom mappings
+	for k, v := range fieldMapping {
+		if strings.HasPrefix(v, "@") {
+			fmt.Fprintf(os.Stderr, "value cannot be prefixed with @: %s", v)
+			os.Exit(1)
+		}
+		mapping[k] = v
+	}
+
+	// add the default mappings
+	for k, v := range j.FieldMapping {
+		mapping[k] = v
+	}
+
+	j.FieldMapping = mapping
+}
+
 // Format implements Formatter.Format to support JSON
 func (j JSONFormatter) Format(w io.Writer, logLevel LogLevel, msg string, timestamp time.Time, fields Fields) error {
 
 	// Standard fields
 	out := Fields{
-		j.getKey("msg"):       msg,
-		j.getKey("timestamp"): getTimestamp(timestamp),
-		j.getKey("level"):     logLevel.String(),
+		"@m": msg,
```

Comment:
I probably missed something here but is is hard-coded with compact?

Reply:
Yes, I thought that since these fields are added no matter what, and they will be unique since it is the only fields that is prefixed with `@` that they could be hard coded? Do you think it should also be defined by a mapping?

Reply:
I fixed so it uses the same the `getKey` func as well.